src/kmerseek/search.py

```diff
@@ -263,11 +265,24 @@ def write_to_file(self, filename):
                 "length",
             ]
         )
-        #
-        # import pdb
-        #
-        # pdb.set_trace()
 
+    def process_results(self, filename):
+        search_results = self.read_results()
+        if self.do_kmer_extraction:
```

Comment:
Passing boolean flags down leads to something like this, where one method has two different behaviors.  Which might be fine in the short term, but what happens when there is a third or fourth?

I wonder if any of the following refactorings might look better?

* Create a new top level command `extract-kmers`
* Create a new `KmerseekExtractKmers` at the same level as `KmerseekIndex`/`KmerseekSearch`
* Create two [specializations](https://en.wikipedia.org/wiki/UML_class_diagram#Generalization/Inheritance) each for `KmerseekIndex`/`KmerseekSearch`, one extracting kmers and one that does not
* Extract the `if self.do_kmer_extraction: ...` and `else:` parts to separate [strategy](https://en.wikipedia.org/wiki/Strategy_pattern) classes, which can be passed in and delegated to